numpy/lib/tests/test_index_tricks.py

```diff
@@ -273,6 +291,19 @@ def test_accepts_npcomplexfloating(self):
             mgrid[0.1:0.3:3j], mgrid[0.1:0.3:np.complex64(3j)]
         )
 
+        # Related to #16945
+        grid64_a = mgrid[0.1:0.3:3.3j]
+        grid64_b = mgrid[0.1:0.3:3.3j,][0]
+        assert_(grid64_a.dtype == grid64_b.dtype == np.float64)
+        assert_array_equal(grid64_a, grid64_b)
+
+        float_type = np.result_type(int, np.longdouble)
+        grid128_a = mgrid[0.1:0.3:np.clongdouble(3.3j)]
+        grid128_b = mgrid[0.1:0.3:np.clongdouble(3.3j),][0]
+        assert_(grid128_a.dtype == grid128_b.dtype == float_type)
```

Comment:
Any reason for this and not
```suggestion
        float_type = np.result_type(int, np.longdouble)
        grid128_a = mgrid[0.1:0.3:np.clongdouble(3.3j)]
        grid128_b = mgrid[0.1:0.3:np.clongdouble(3.3j),][0]
        assert_(grid128_a.dtype == grid128_b.dtype == np.longdouble)
```

Reply:
This is specifically for windows: 
`np.result_type(int, np.longdouble) is np.float64` as opposed to `... is np.longdouble`.

Reply:
Not for me:
```
In [173]: np.result_type(int, np.longdouble) is np.float64
Out[173]: False

In [174]: np.__version__
Out[174]: '1.17.4'
```
Did this change recently?

Reply:
Ah, correction: it returns `dtype(float64)`, not just `float64`:

``` python
In [1]: import numpy as np

In [2]: np.result_type(int, np.longdouble)
Out[2]: dtype('float64')
```

Reply:
```
In [1]: np.result_type(int, np.longdouble)
Out[1]: dtype('float64')

In [2]: np.result_type(int, np.longdouble).type is np.float64
Out[2]: False
```
Can you confirm that on master that `False` is `True`?

Reply:
> Can you confirm that on master that the `False` and `True` swap?

False alarm. the string notation of `np.dtype(np.longlong)` was the cause of confusion here.
The `np.result_type(int, np.longdouble).type is np.longdouble` expression indeed holds for the master, even for windows.

Reverted in https://github.com/numpy/numpy/pull/16971/commits/e548b9a7812607c12ab6afa5a7f659484d29af69.